src/Features/LanguageServer/ProtocolUnitTests/SemanticTokens/SemanticTokensTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Xunit;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests.SemanticTokens
+{
+    public class SemanticTokensTests : AbstractSemanticTokensTests
+    {
+        [Fact]
+        public async Task TestGetSemanticTokensAsync()
```

Comment:
Can we add a test somewhere that invokes all the requests in the expected order (e.g. range, full document, edit1, edit2 and makes sure we get the expected result at each stage?

Reply:
Might also be worth checking that you can interleave full document requests in the middle there too. In the case of erroneous or hard to apply edits, the client may re-request the full document.